mula/scheduler/storage/storage.py

```diff
@@ -1,68 +1,67 @@
 import json
-import logging
-import time
-from functools import partial, wraps
+from functools import partial
 
 import sqlalchemy
+import structlog
 
 from scheduler.config import settings
 
-logger = logging.getLogger(__name__)
+from .errors import StorageError
 
 
 class DBConn:
-    def __init__(self, dsn: str) -> None:
-        super().__init__()
+    def __init__(self, dsn: str, pool_size: int = 25) -> None:
+        self.logger: structlog.BoundLogger = structlog.getLogger(__name__)
 
-        self.engine = None
+        self.dsn: str = dsn
+        self.pool_size: int = pool_size
```

Comment:
The type hints here are redundant, they can be inferred (by users and type checkers) from type hints given in the signature